### PR TITLE
Allow running browser instance for web extension testing

### PR DESCRIPTION
**Pull request by christian-bromann**
Currently the service only allows to test the extension as app extension but many extensions are also served through the web browser via e.g. codespaces. The service should allow to specify running the tests in the browser. For that we should take a look at what [`@vscode/test-web`](https://github.com/microsoft/vscode-test-web) does and either port its functionality over or use the project as dependency (which is currently a no-go as it has playwright as dependency).

### CODE REVIEW

File: src/launcher.ts [hunk 1]
```diff
@@ -123,9 +189,27 @@ export default class VSCodeServiceLauncher extends ChromedriverServiceLauncher {
...
+            if (version === 'main') {
+                log.info(`Fetch latest GitHub release from ${VSCODE_LATEST_RELEASE}`)
+                const { body: versions } = await request(VSCODE_LATEST_RELEASE, {
+                    headers: {
+                        Accept: 'application/vnd.github.v3+json',
+                        'User-Agent': 'webdriverio-community/wdio-vscode-service'
+                    }
+                })
+                const response: { tag_name: string } = await versions.json()
+                version = response.tag_name
```
Comment (mattseddon) [inline src/launcher.ts:210]: @christian-bromann I just tested with v4. I have my config set with:
```
  capabilities: [
    {
      browserName: 'vscode',
      browserVersion: 'insiders',
...
```
but the stable version `1.69.2` is being returned. Does the above request include the insiders version or only stable releases?

Comment (mattseddon) [reply src/launcher.ts:210]: Or am I doing something wrong on my end?

File: src/pageobjects/workbench/Input.ts [hunk 2]
```diff
@@ -34,18 +37,18 @@ export abstract class Input extends BasePage<AllInputLocators> {
...
 
         // fallback to clipboard if the text gets malformed
-        if ((await this.getText()) !== text) {
+        const currentText = await this.getText()
+        if (currentText !== text) {
             await clipboard.write(text)
-            await input.addValue(['End'])
-            await input.addValue(['Shift', 'Home'])
-            await input.addValue([CMD_KEY, 'v'])
+            const backSpaces = new Array(currentText.length).fill('Backspace')
```
Comment (mattseddon) [inline src/pageobjects/workbench/Input.ts:50]: Looks like we can in here with `currentText` as null. Could use:

```
            if (currentText?.length){
                const backSpaces = new Array(currentText.length).fill('Backspace');
                await input.addValue(backSpaces);
            }
```

File: src/launcher.ts (hunk 1 above)
Comment (christian-bromann) [reply src/launcher.ts:210]: let me check

**Pull request merged**